### ISSUE TITLE
Native CPU is not reported on Raspbian

### ISSUE THREAD

**Issue by kryptan** [rust-lang/rust#84890]
Executing `rustc --print target-cpus` or `rustc -Ctarget-cpu=help` on Raspberry Pi 4 prints the following:

```
Available CPUs for this target:
    arm1020e
    arm1020t
    arm1022e
    ...
```

In contrast, executing the same command on Windows or Ubuntu on x86-64 also prints native CPU:

```
Available CPUs for this target:
    native         - Select the CPU of the current host (currently skylake).
    alderlake
    amdfam10
    athlon
    ...
```

In [source code](https://github.com/rust-lang/rust/blob/fed59d669c5ca3c0e9c39dcb1f6510b5876ede64/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp#L397) there is check for cross compilation. Native CPU is only printed when not cross compiling:

```C++
  printf("Available CPUs for this target:\n");
  if (HostArch == TargetArch) {
    const StringRef HostCPU = sys::getHostCPUName();
    printf("    %-*s - Select the CPU of the current host (currently %.*s).\n",
      MaxCPULen, "native", (int)HostCPU.size(), HostCPU.data());
  }
```

However I'm not cross compiling. I'm using rust installed from rustup and don't have any additional targets installed.

`rustup show`:
```
Default host: armv7-unknown-linux-gnueabihf
rustup home:  /home/pi/.rustup

nightly-armv7-unknown-linux-gnueabihf (default)
rustc 1.54.0-nightly (716394d65 2021-05-03)
```

`lsb_release -a`:
```
No LSB modules are available.
Distributor ID: Raspbian
Description:    Raspbian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
```

`rustc --version --verbose`:
```
rustc 1.54.0-nightly (716394d65 2021-05-03)
binary: rustc
commit-hash: 716394d6581b60c75cfdd88b8e5b876f2db88b62
commit-date: 2021-05-03
host: armv7-unknown-linux-gnueabihf
release: 1.54.0-nightly
LLVM version: 12.0.0
```

Relevant #44393 #44407

**Comment by SvenDowideit** [comment]
Is this related to what I hit in https://github.com/rust-rse/reed-solomon-erasure/issues/94 ?

on a coretex-a57 SoftIron overdrive1000 from 2016 ...

```
# rustc --version --verbose
rustc 1.59.0 (9d1b2106e 2022-02-23)
binary: rustc
commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
commit-date: 2022-02-23
host: aarch64-unknown-linux-gnu
release: 1.59.0
LLVM version: 13.0.0
```
